src/oracles/CoinbaseOracle.sol

```diff
@@ -0,0 +1,120 @@
+     * @param _decimals The number of decimals in the reported price.
+     * @param _reporter The address of the reporter allowed to push price data.
+     */
+    constructor(string memory _symbol, uint8 _decimals, address _reporter) {
+        symbol = _symbol;
+        decimals = _decimals;
+        reporter = _reporter;
+    }
+
+    /**
+     * @notice Returns the latest round data.
+     * @return roundId The ID of the latest round.
+     * @return answer The latest price.
+     * @return startedAt The timestamp when the round started.
+     * @return updatedAt The timestamp when the round was updated.
+     * @return answeredInRound The round ID in which the answer was computed.
+     */
+    function latestRoundData()
+        external
+        view
+        override
+        returns (uint80 roundId, int256 answer, uint256 startedAt, uint256 updatedAt, uint80 answeredInRound)
+    {
+        int256 latestAnswer = roundAnswers[lastRoundId];
+        uint256 latestTimestamp = roundTimestamps[lastRoundId];
+        return (lastRoundId, latestAnswer, latestTimestamp, latestTimestamp, lastRoundId);
+    }
+
+    /**
+     * @notice Returns the data for a specific round.
+     * @param _roundId The round ID to retrieve the data for.
+     * @return roundId The ID of the round.
+     * @return answer The price of the round.
+     * @return startedAt The timestamp when the round started.
+     * @return updatedAt The timestamp when the round was updated.
+     * @return answeredInRound The round ID in which the answer was computed.
+     */
+    function getRoundData(uint80 _roundId)
+        external
+        view
+        override
+        returns (uint80 roundId, int256 answer, uint256 startedAt, uint256 updatedAt, uint80 answeredInRound)
+    {
+        int256 latestAnswer = roundAnswers[_roundId];
+        uint256 latestTimestamp = roundTimestamps[_roundId];
+        return (_roundId, latestAnswer, latestTimestamp, latestTimestamp, _roundId);
+    }
+
+    /**
+     * @notice Pushes a new price to the oracle.
+     * @param priceData The encoded price data.
+     * @param signature The signature to verify the authenticity of the data.
+     */
+    function pushPrice(bytes memory priceData, bytes memory signature) external {
+        (
+            string memory kind, // e.g. "price"
+            uint256 timestamp, // e.g. 1629350000
+            string memory ticker, // e.g. "BTC"
+            uint256 price // 6 decimals
+        ) = abi.decode(priceData, (string, uint256, string, uint256));
+
+        uint256 latestTimestamp = roundTimestamps[lastRoundId];
+
+        require(keccak256(abi.encodePacked(kind)) == keccak256(abi.encodePacked("price")), "Invalid kind.");
+        require(price < uint256(type(int256).max), "Price exceeds max value.");
+        require(timestamp > latestTimestamp, "Invalid timestamp.");
+        require(keccak256(abi.encodePacked(ticker)) == keccak256(abi.encodePacked(symbol)), "Invalid ticker.");
```

Comment:
one implication with the design here is we need one instance of `CoinbaseOracle` deployed for each `symbol` we want to support. If we have 10 of these, for example, there will be 10 contracts we need to then update with each coinbase update.

two other options are:
1) update this contract to store multiple `symbol`s price and update time
2) create a wrapper contract (can be done later) that you can call into with the batch update which then calls into the `CoinbaseOracle` contract and pushes the associated prices

(2) is likely better, but it is more expensive as there is an extra intra-contract call for each `symbol` we support.

I think for now we can leave it like this and pretty easily add this wrapper later (it could even be calling into a standard DSProxy with an array of payloads).